### PR TITLE
005 job role info

### PR DESCRIPTION
**Pull request by Conaill4**

### CODE REVIEW

File: src/services/JobRoleService.ts [hunk 1]
```diff
@@ -14,3 +15,14 @@ export const getJobs = async (): Promise<JobRole[]> => {
         throw new Error('Failed to get any Jobs')
     }
 }
+
+export const getJobByID_ = async (id: string): Promise<JobRoleDetailedResponse[]> => {
```
Comment (mlynch147) [inline src/services/JobRoleService.ts]: can you rename this to getJobDetailsById

File: src/validators/JobRoleDetailedResponseValidator.ts
Comment (mlynch147) [inline src/validators/JobRoleDetailedResponseValidator.ts:1]: if this is not called anywhere in the code (tests excluded) - then please remove....

File: views/job-role-list.html [hunk 2]
```diff
@@ -107,14 +108,14 @@ <h4 style="text-align: left;">The following jobs are available at Kainos: &nbsp;
...
-                                                                                    <td class="td_data" id="identification"><a href="/job-roles/{{JobRole.jobRoleId}}">{{ JobRole.jobRoleId }}</a></td>
-                                                                                    <td class="td_data" id="name"><a href="/job-roles/{{JobRole.jobRoleId}}">{{ JobRole.roleName }}</a></td>
-                                                                                    <td class="td_data" id="location">{{ JobRole.location }}</td>
-                                                                                    <td class="td_data" id="capabilityName">{{ JobRole.capabilityName }}</td>
-                                                                                    <td class="td_data" id="bandName">{{ JobRole.bandName }}</td>
-                                                                                    <td class="td_data"id="closingDate">{{ JobRole.closingDate | date }}</td>
+                                                                                    <td class="td_data"><a href="/job-roles/{{JobRole.jobRoleId}}">{{ JobRole.jobRoleId }}</a></td>
```
Comment (mlynch147) [inline views/job-role-list.html]: re-add any ids that are needed for testing...

File: views/job-role-list.html [hunk 3]
```diff
@@ -107,14 +108,14 @@ <h4 style="text-align: left;">The following jobs are available at Kainos: &nbsp;
...
+                                                                                    <td class="td_data"><a href="/job-roles/{{JobRole.jobRoleId}}">{{ JobRole.roleName }}</a></td>
+                                                                                    <td class="td_data"><a target="_blank" href="https://www.google.com/maps/place/{{ JobRole.location }}">{{ JobRole.location }}</a></td>
```
Comment (mlynch147) [inline views/job-role-list.html]: remove - not required...

File: views/job-role-list.html [hunk 4]
```diff
@@ -107,14 +108,14 @@ <h4 style="text-align: left;">The following jobs are available at Kainos: &nbsp;
                                                                             <th class="th_data">Band:</th>
                                                                             <th class="th_data">Closing Date:</th>
                                                                         </tr>
-                                                                        {% for JobRole in JobRoles | sort(attribute='JobRoleId') %}
+                                                                        {% for JobRole in JobRoles | sort(attribute='jobRoleId') %}
                                                                             <tr >
-                                                                                    <td class="td_data" id="identification"><a href="/job-roles/{{JobRole.jobRoleId}}">{{ JobRole.jobRoleId }}</a></td>
```
Comment (mlynch147) [inline views/job-role-list.html:112]: add unique ids in to any fields that are required....

**Review by amarauzoma — APPROVED** [review]

**Review by mlynch147 — APPROVED** [review]

File: views/home.html [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
+    {% include "components/header.html" %} 
+    <title>Home - Kainos</title>  
 </head>
 <body class="static-page Life at Kainos" style="visibility: hidden;">
 
     <div class="main-header">
         {% include "components/navigationBar.html" %}    
    </div>
 
+   {% if errormessage %}
```
Comment (Conaill4) [inline views/home.html]: remove this block

File: src/services/JobRoleService.ts [hunk 6]
```diff
@@ -14,3 +15,15 @@ export const getJobs = async (): Promise<JobRole[]> => {
...
 }
+
+export const getJobDetailsById = async (id: string): Promise<JobRoleDetailedResponse[]> => {
+    try {
+        const response: AxiosResponse = await axios.get(URL+id)
+
+        return response.data;
+    } catch (e) {
+        
+        console.log(e);
```
Comment (Conaill4) [inline src/services/JobRoleService.ts]: remove this log

**Pull request merged**